### ISSUE TITLE
Deciphering scores of individual choices within a trace? 

### ISSUE THREAD

**Issue by collinskatie** [probcomp/Gen.jl#347]
Hi, 

I was wondering if there are any recommended ways to programmatically extract the Gen.ChoiceOrCall structure from a trace, or something like it? When I print out a trace, I see all the choices that were made, as well as what I'm fairly sure from looking at the documentation is the penalty to the score incurred from each choice (?) 

Is there a recommended way to extract for each trace, the penalty associated with a given choice, say :n if :n represents a sample from a distribution over the number of blocks in a scene? Like `n = @trace(geometric(sparsity_param), :n);` - as one call within many to generate a single trace. 

In other words, I see how trace[:n] will extract the sampled number, but it would be great to be able to easily access the score penalty of that choice towards the entire trace - is there already a function that does this? 

For reference, this is what I see for a sample trace for that address: `(:n => Gen.ChoiceOrCallRecord{Int64}(2, -9.220390707829683, NaN, true)` and it would be great to easily extract this structure, and the second element of it, if that is at all possible (or meaningful to investigate in the first place). 

Thank you for any help!

**Comment by marcoct** [comment]
@collinskatie You can use `Gen.project` for this, e.g.:

```julia
Gen.project(trace, select(:n))
```

**Comment by collinskatie** [comment]
Awesome! Thank you!

**Comment by collinskatie** [comment]
@marcoct thanks again for pointing out that function, which worked! I have two follow-up questions related to Gen.project and scores in general:

1) When running Gen.project(trace, select(:choice_addr)) -- is there a reasonable interpretation for why that may return a very positive value for a given choice? I know the following is the function definition, but say :choice_addr gets a projected score of +30, but the overall trace has a Gen.get_score(trace) of something like -100, does that imply that the sampled value(s) for :choice_addr are very "good"? 

<img width="811" alt="Screen Shot 2021-01-23 at 11 28 38 PM" src="https://user-images.githubusercontent.com/37914552/105621228-50340080-5dd3-11eb-9455-6d39c2b07b15.png">

2) In our model, we draw 20 samples for possible collision values before calling a Python-based physics engine. On each simulation, the model often does not use all 20 collision values; however, all 20 draws from the distribution are included in the score of the trace (even if unused). Is there a way to "mask out" or remove some traced choices during inference or calls to Gen.get_score for trace? Not sure if this is possible since the underlying noise is used in the Python module, not Gen/Julia - but just wanted to check. 

Thanks!
